### PR TITLE
Add Fidelity keyboard 

### PR DESCRIPTION
**Pull request by mini-ninja-64**
## Description

Helllooo

This adds the Fidelity keyboard to the QMK repo.
I tried to use the new `info.json` configuration where possible

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/cutie_club/fidelity/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    /* Base */
+    [0] = LAYOUT(
+        KC_ESC,      KC_1,        KC_2,        KC_3,         KC_4,       KC_5,        KC_6,        KC_7,        KC_8,        KC_9,        KC_0,        KC_MINS,      KC_EQL,     KC_BSPC, KC_BSPC, KC_DEL,
+        KC_TAB,      KC_Q,        KC_W,        KC_E,         KC_R,       KC_T,        KC_Y,        KC_U,        KC_I,        KC_O,        KC_P,        KC_LBRC,      KC_RBRC,    KC_BSLS,          KC_PGUP,
+        KC_CAPS,     KC_A,        KC_S,        KC_D,         KC_F,       KC_G,        KC_H,        KC_J,        KC_K,        KC_L,        KC_SCLN,     KC_QUOT,      KC_NUHS,    KC_ENT,           KC_PGDN,
+        KC_LSFT,     KC_NUBS,     KC_Z,        KC_X,         KC_C,       KC_V,        KC_B,        KC_N,        KC_M,        KC_COMM,     KC_DOT,      KC_SLSH,      KC_RSFT,             KC_UP,   MO(1),
```
Comment (zvecr) [inline keyboards/cutie_club/fidelity/keymaps/default/keymap.c:25]: `MO(1)` here would go out of bounds.

File: keyboards/cutie_club/fidelity/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    ),
+    [3] = LAYOUT(
+        KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,      KC_TRNS,    KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,      KC_TRNS,    KC_TRNS,          KC_TRNS,
+        KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,      KC_TRNS,    KC_TRNS,          KC_TRNS,
+        KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,      KC_TRNS,             KC_TRNS, KC_TRNS,
+        KC_TRNS,                  KC_TRNS,                                                         KC_TRNS,                                                          KC_TRNS,    KC_TRNS, KC_TRNS, KC_TRNS
+    )
```
Comment (zvecr) [inline keyboards/cutie_club/fidelity/keymaps/via/keymap.c:48]:
```suggestion
    )
```

File: keyboards/cutie_club/fidelity/readme.md
Comment (zvecr) [inline keyboards/cutie_club/fidelity/readme.md:1]: This file should follow the [template](https://github.com/qmk/qmk_firmware/blob/master/data/templates/keyboard/readme.md).

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/cutie_club/fidelity/keymaps/default/keymap.c (hunk 1 above)
Comment (mini-ninja-64) [reply keyboards/cutie_club/fidelity/keymaps/default/keymap.c:25]: My apologies, I missed this and have now rectified the issue

File: keyboards/cutie_club/fidelity/readme.md
Comment (mini-ninja-64) [reply keyboards/cutie_club/fidelity/readme.md:1]: I have updated to now follow the template.

File: keyboards/cutie_club/fidelity/keymaps/via/keymap.c (hunk 2 above)
Comment (mini-ninja-64) [reply keyboards/cutie_club/fidelity/keymaps/via/keymap.c:48]: I was under the impression that, VIA keymaps should pre-allocate the number of dynamic layers the keyboard wants to support, so if I want to allow users 4 layers in VIA I would need 4 layers allocatied in the VIA keymap.

It is very likely I am misunderstanding, I just want to check?

Comment (zvecr) [reply keyboards/cutie_club/fidelity/keymaps/via/keymap.c:48]: > VIA keymaps should pre-allocate the number of dynamic layers the keyboard wants to support,

This is controlled with `DYNAMIC_KEYMAP_LAYER_COUNT`, not via the method you have attempted to use.

Padding the keymap array for init purposes is also no longer required, as unspecified layers will default to `KC_TRANS`.

Comment (mini-ninja-64) [reply keyboards/cutie_club/fidelity/keymaps/via/keymap.c:48]: Gotya, that makes sense, thank you for the explination (o˘◡˘o)

**Review by keyboard-magpie — APPROVED** [review]

File: keyboards/cutie_club/fidelity/readme.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# Fidelity
+
+![Fidelity](https://i.imgur.com/6xelfJe.jpeg)
```
Comment (drashna) [inline keyboards/cutie_club/fidelity/readme.md:3]:
```suggestion
![Fidelity](https://i.imgur.com/6xelfJeh.jpeg)
```

**Review by drashna — APPROVED** [review]

File: keyboards/cutie_club/fidelity/readme.md (hunk 3 above)
Comment (mini-ninja-64) [reply keyboards/cutie_club/fidelity/readme.md:3]: Apologies, I missed that, thank you for the correction \(★ω★)/

**Comment by mini-ninja-64** [comment]
Thank you for your review, @zvecr does this look ok now?

**Pull request merged**